extensions/blocks/donations/donations.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Donations Block.
+ *
+ * @since 8.x
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Donations;
+
+use Jetpack_Gutenberg;
+
+const FEATURE_NAME = 'donations';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	jetpack_register_block(
+		BLOCK_NAME,
+		array( 'render_callback' => __NAMESPACE__ . '\load_assets' )
+	);
```

Comment:
Is the block going to be available to all users, or are you planning on restricting it to some specific plans? If the latter, you'd want to add a plan check here:
https://github.com/Automattic/jetpack/blob/5a4ec0da4f32874708b4f7014690aa458ceca883/extensions/README.md#L157

Reply:
We want the block to be available in all plans, but usable only on paid plans. Users on a free plan can insert the block but they will see an upgrade nudge:

https://github.com/Automattic/jetpack/blob/1a8ca1cd66aaee9b711e1bb15a65ee7b113a5a84/extensions/blocks/donations/edit.js#L96-L98

Due to that, we want the block to be always registered. AFAIK we cannot use the `plan_check => true` arg because that will hide the block from the inserter. The `jetpack_block_editor_enable_upgrade_nudge` filter could help displaying the block in the inserter, but seems it doesn't currently offer enough granularity so we can display only this block without affecting the rest of paid blocks that are not currently shown in the inserter.